src/plugins/teleport-plugin-react-styled-components/index.ts

```diff
@@ -0,0 +1,69 @@
+import { ComponentPluginFactory, ComponentPlugin } from '../../typings/generators'
+import { generateStyledComponent } from './utils'
+import { traverseElements, transformDynamicStyles } from '../../shared/utils/uidl-utils'
+import { stringToUpperCamelCase } from '../../shared/utils/string-utils'
+import { createJSXSpreadAttribute } from '../../shared/utils/ast-jsx-utils'
+
+interface StyledComponentsConfig {
+  componentChunkName: string
+  importChunkName?: string
+}
+
+export const createPlugin: ComponentPluginFactory<StyledComponentsConfig> = (config) => {
+  const { componentChunkName = 'react-component', importChunkName = 'import-local' } = config || {}
+
+  const reactStyledComponentsPlugin: ComponentPlugin = async (structure) => {
+    const { uidl, chunks, dependencies } = structure
+    const { node } = uidl
+    const componentChunk = chunks.find((chunk) => chunk.name === componentChunkName)
+    if (!componentChunk) {
+      return structure
+    }
+
+    const jsxNodesLookup = componentChunk.meta.nodesLookup
+    const jssStyleMap = {}
+
+    traverseElements(node, (element) => {
+      const { style, key, elementType } = element
+      if (style) {
+        const root = jsxNodesLookup[key]
+        const className = `${stringToUpperCamelCase(key)}Wrapper`
+        let needInjectionOfProps: boolean = false
+        jssStyleMap[className] = transformDynamicStyles(style, (styleValue) => {
+          if (styleValue.content.referenceType === 'prop') {
+            needInjectionOfProps = true
+            return `\$\{props => props.${styleValue.content.id}\}`
+          }
+          throw new Error(
+            `Error running transformDynamicStyles in reactStyledComponentsPlugin. Unsupported styleValue.content.referenceType value ${
+              styleValue.content.referenceType
+            }`
+          )
+        })
```

Comment:
more of a styling thing, try to leave an empty line after each if block, so the code looks better separated. I'll have to see if prettier/tslint have any specific setting for this